WordPress/gutenberg
# Update control labels to the new uppercase styles

Closes #42782

## What?

Updates all control component labels to the new uppercase styles.

## Why?

Most of the missing font-size declarations discovered in #42781 were in control labels, so this is a good opportunity to do a sweep and update.

## How?

The label styles in our repo currently originate from either `InputControl` or `BaseControl`. Although we should eventually merge them completely, as an incremental step in this PR I've extracted the shared typography styles between the two into a low-level mixin called `baseLabelTypography`.

Next, all components that have ad hoc styled labels have been replaced with BaseControl's `StyledLabel` or `BaseControl.VisualLabel`. The two are basically the same, except that `StyledLabel` is `<label>` by default and is not publicly exported, while `BaseControl.VisualLabel` is `<span>`  by default and publicly exported. Both are polymorphic. We should streamline this more, but there are also some TypeScript complications so I'm not going to do it in this PR.

## Testing Instructions

1. `npm run storybook:dev` and check all "control" components.
2. `npm run dev` and check the components in the post/site editors.

### Out of scope for this PR

- #42788
- (maybe worth looking into) #34609
- The `ControlLabel`/`FormGroup` components are left untouched because they aren't publicly exported or used anywhere.
- InputControl: The `size` prop is being incorrectly passed through to a `Text`-based label that results in an invalid `font-size` value. This isn't making a visual difference because the value is overridden anyway, but I'll fix it separately in #42793.

## Screenshots or screencast 

https://user-images.githubusercontent.com/555336/181645544-ffbd74bf-bfbe-4bd9-a9ad-c9f2d498f368.mp4
